arangod/Aql/ClusterQuery.cpp

```diff
@@ -97,7 +97,7 @@ std::shared_ptr<ClusterQuery> ClusterQuery::create(
 
 void ClusterQuery::prepareClusterQuery(
     VPackSlice querySlice, VPackSlice collections, VPackSlice variables,
-    VPackSlice snippets, VPackSlice traverserSlice, std::string_view user,
+    VPackSlice snippets, VPackSlice traverserSlice, std::string const& user,
```

Comment:
I do not understand why this helps. If the UB is that the `user` `string_view` pointed to memory which was later no longer valid, then using a `std::string const&` does not help. I looked at the call sites of this function and found one place, where `ExecContext::current().user()` was used, which returns a **reference** to a string in the `ExecContext`. Now this is thread local and in the new era of coroutines, it is possible that the current `ExecContext` is different, when the coroutine is resumed in a different thread. I fear we must make this argument a `std::string` by value and then move it to its destination to fix the UB.

Reply:
This may be a possible issue in devel. But in 3.10 we don't use coroutines at all.

Reply:
>  I fear we must make this argument a std::string by value and then move it to its destination to fix the UB.

It also can be better because for now code copying string under mutex lock

Reply:
I wouldn't care about the copying - this is the username which in virtually all cases will be short enough to fit in the small buffer, so it doesn't matter whether we move or copy, we have to do the same work.